update geocoder

datendelphin opened the pull request:
search no longer worked. The problem is that the geocoder thing loads 
https://nominatim.openstreetmap.org/search/?q=...
But nominatim API is without the slash after search:
https://nominatim.openstreetmap.org/search?q=... works

solution: update leaflet-control-geocoder

yuryleb commented on package.json:
```diff
@@ -28,19 +28,20 @@
     "uglify-js": "^2.8.0"
   },
   "dependencies": {
+    "corslite": "^0.0.7",
```
Is this new dependency really necessary?

datendelphin replied:
I don't know. I think I arrived at this via npm audit. And previously I always had `npm install corslite` as an additional step, without it would never build. But I have very little experience with npm and javascript in general, so I can just remove it if you feel that would be better.

mstock committed:
Update various NPM dependencies to current versions

This also required a change to the `uglify` command whose `--source-map`
command line parameter apparently changed its behaviour.

mstock committed:
Adapt to changed `leaflet-control-geocoder` behaviour

Apparently, the object returned by requiring `leaflet-control-geocoder`
appears to be an empty object, so calling a function on it was failing.
It can be accessed via the `L.Control.Geocoder` object though, so take
that route.

mstock committed:
Pass itinerary builder as option to control

Otherwise, this wasn't working anymore and just displayed `[object
Object]` instead of a useful instruction. This approach also doesn't
overwrite the original `ItineraryBuilder` class of the
`leaflet-routing-machine` anymore, which feels a bit cleaner, too.

datendelphin committed:
suggestions from npm audit

datendelphin commented:
Apparently corslite it is really not needed, my bad. (needed in the actually used version on the demo server which has a shorturl service)

yuryleb commented on package.json:
```diff
@@ -6,7 +6,7 @@
     "test": "eslint src/*.js i18n/*.js",
     "replace": "node ./scripts/replace.js",
-    "compile": "browserify -d src/index.js -s osrm > bundle.raw.js && uglifyjs bundle.raw.js -c -m --source-map=bundle.js.map -o bundle.js",
+    "compile": "browserify -d src/index.js -s osrm > bundle.raw.js && uglifyjs bundle.raw.js -c -m --source-map filename=bundle.js.map -o bundle.js",
```
I'm not sure `--source-map filename=bundle.js.map` option should contain whitespace, maybe dash instead?

datendelphin replied:
Well see the full comment on commit https://github.com/Project-OSRM/osrm-frontend/pull/370/commits/6d175909a5f1f7dfb01851223057ab5680042f4b

Apparently the command line behaviour changed. --source-map now takes a string of key value pairs. I found https://stackoverflow.com/questions/15590702/how-to-get-minified-output-with-browserify where someone uses three key-value pairs for that argument.

nclslbrn commented:
I just fixed it by removing the slash on geocode.js (line 373) 

Before `L.Control.Geocoder.jsonp( this.options.serviceUrl + 'search/'...`
After `L.Control.Geocoder.jsonp( this.options.serviceUrl + 'search'...`

```javascript

geocode( query, cb, context ) {
	L.Control.Geocoder.jsonp( this.options.serviceUrl + 'search', L.extend( {
		q: query,
		limit: 5,
		format: 'json',
		addressdetails: 1,
	}, this.options.geocodingQueryParams ),
	function( data ) {
		const results = [];
		for ( let i = data.length - 1; i >= 0; i-- ) {
			const bbox = data[ i ].boundingbox;
			for ( let j = 0; j < 4; j++ ) {
				bbox[ j ] = parseFloat( bbox[ j ] );
			}
			results[ i ] = {
				icon: data[ i ].icon,
				name: data[ i ].display_name,
				html: this.options.htmlTemplate
					? this.options.htmlTemplate( data[ i ] )
					: undefined,
				bbox: L.latLngBounds( [ bbox[ 0 ], bbox[ 2 ] ], [ bbox[ 1 ], bbox[ 3 ] ] ),
				center: L.latLng( data[ i ].lat, data[ i ].lon ),
				properties: data[ i ],
			};
		}
		cb.call( context, results );
	}, this, 'json_callback' );
},


```
